src/app/user/admin.guard.ts

```diff
@@ -0,0 +1,16 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, RouterStateSnapshot, UrlTree } from '@angular/router';
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AdminGuard implements CanActivate {
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    return true;
```

Comment:
Implement admin.guard.ts which should follow CanActivate interface and should return true in case if user is admin and redirect to /courses in case if user is not admin with help of UrlTree.

Reply:
admin.guard.ts for /course/add and /course/edit/:id routes